--- a/src/vesa.h
+++ b/src/vesa.h
@@ -60,7 +60,9 @@
 /* Int 10 support */
 #include "xf86int10.h"
 
+#ifdef XFreeXDGA
 /* Dga definitions */
 #include "dgaproc.h"
+#endif
 
 #include "fb.h"
 
@@ -103,10 +105,12 @@ typedef struct _VESARec
     int			maxClock;
     MessageType		clockRangeSource;
 
+#ifdef XFreeXDGA
     /* DGA info */
     DGAModePtr pDGAMode;
     int nDGAMode;
     CloseScreenProcPtr CloseScreen;
+#endif
     xf86MonPtr monitor;
 
     OptionInfoPtr Options;
--- a/src/vesa.c
+++ b/src/vesa.c
@@ -65,6 +65,7 @@ static const OptionInfoRec * VESAAvailableOptions(int chipid, int busid);
 static void	VESAIdentify(int flags);
 static Bool	VESAProbe(DriverPtr drv, int flags);
 static Bool	VESAPreInit(ScrnInfoPtr pScrn, int flags);
+#ifdef XFreeXDGA
 static Bool	VESADGAInit(ScrnInfoPtr pScrn, ScreenPtr pScreen);
+#endif
 
 static Bool	VESAEnterVT(ScrnInfoPtr pScrn);
 static void	VESALeaveVT(ScrnInfoPtr pScrn);
@@ -503,7 +504,9 @@ VESAPreInit(ScrnInfoPtr pScrn, int flags)
     }
 
     pScrn->currentMode = pScrn->modes;
+#ifdef XFreeXDGA
     pVesa->nDGAMode = 0;
+#endif
 
     pScrn->displayWidth = pScrn->virtualX;
 
@@ -817,7 +820,9 @@ VESAScreenInit(ScreenPtr pScreen, int argc, char **argv)
     if (!VESACloseScreen(pScreen))
 	return FALSE;
 
+#ifdef XFreeXDGA
     VESADGAInit(pScrn, pScreen);
+#endif
 
     xf86SetBackingStore(pScreen);
     xf86SetSilkenMouse(pScreen);
@@ -890,11 +895,13 @@ VESACloseScreen(ScreenPtr pScreen)
     ScrnInfoPtr pScrn = xf86ScreenToScrn(pScreen);
     VESAPtr pVesa = VESAGetRec(pScrn);
 
+#ifdef XFreeXDGA
     if (pVesa->pDGAMode) {
 	free(pVesa->pDGAMode);
 	pVesa->pDGAMode = NULL;
 	pVesa->nDGAMode = 0;
     }
+#endif
     if (pScrn->vtSema) {
 	VESARestore(pScrn);
 	VESAUnmapVidMem(pScrn);
@@ -996,6 +1003,7 @@ VESAValidMode(ScrnInfoPtr pScrn, DisplayModePtr pMode, Bool verbose, int flags)
     return MODE_OK;
 }
 
+#ifdef XFreeXDGA
 /*
  * DGA stuff
  */
@@ -1222,3 +1230,4 @@ VESADGAInit(ScrnInfoPtr pScrn, ScreenPtr pScreen)
     return (DGAInit(pScreen, &VESADGAFunctions,
 	    pVesa->pDGAMode, pVesa->nDGAMode));
 }
+#endif
